index.d.ts

```diff
@@ -369,6 +370,36 @@ export interface PlanFn {
 	skip(count: number): void;
 }
 
+export interface TryFn<Context> {
+	<T extends any[]>(
+		title: string,
+		impl: (t: ExecutionContext<Context>, ...args: T) => ImplementationResult,
```

Comment:
I'd like to see if we could type this to accept a macro type. Happy to try that out myself.

Reply:
Sounds good